File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    circle = elements * 2  
    answer = set()

    for length in range(1, n + 1):
        for start in range(n):
            temp = sum(circle[start:start + length])
            answer.add(temp)

    return len(answer)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int n=elements.length;
        HashSet<Integer> set=new HashSet<>();
        
        int[] array=new int[n*2];
        
        for(int i=0;i<2*n;i++){
            array[i]=elements[i%n];
        }
        
        for(int length=1;length<=n;length++){
            for(int start=0;start<n;start++){
                int sum=0;
                for(int i=0;i<length;i++){
                    sum+=array[start+i];
                }
                set.add(sum);
            }
        }
        
        return set.size();
    }
}